libdnf/base/goal.cpp

```diff
@@ -657,25 +678,29 @@ void Goal::Impl::add_upgrades_distrosync_to_goal(Action action, const std::strin
         return;
     }
     // Report when package is not installed
-    if (!utils::is_glob_pattern(spec.c_str()) && !nevra_pair.second.get_name().empty()) {
-        // Report only not installed if not obsoleters - https://bugzilla.redhat.com/show_bug.cgi?id=1818118
-        rpm::SolvQuery all_installed(&sack);
-        all_installed.ifilter_installed();
-        bool obsoleters = false;
-        if (obsoletes) {
-            rpm::SolvQuery obsoleters_query(query);
-            obsoleters_query.ifilter_obsoletes(all_installed);
-            if (!obsoleters_query.empty()) {
-                obsoleters = true;
-            }
+    rpm::SolvQuery all_installed(&sack, rpm::SolvQuery::InitFlags::IGNORE_EXCLUDES);
+    all_installed.ifilter_installed();
+    // Report only not installed if not obsoleters - https://bugzilla.redhat.com/show_bug.cgi?id=1818118
+    bool obsoleters = false;
+    if (obsoletes && action != Action::DOWNGRADE) {
+        rpm::SolvQuery obsoleters_query(query);
+        obsoleters_query.ifilter_obsoletes(all_installed);
+        if (!obsoleters_query.empty()) {
+            obsoleters = true;
         }
-        if (!obsoleters) {
-            all_installed.ifilter_name({nevra_pair.second.get_name()});
-            if (all_installed.empty()) {
-                add_rpm_goal_report(action, GoalProblem::NOT_INSTALLED, settings, spec, {}, false);
-            } else if (all_installed.ifilter_arch({nevra_pair.second.get_name()}).empty()) {
+    }
+    rpm::SolvQuery relevant_installed_na(all_installed);
+    if (!obsoleters) {
+        relevant_installed_na.ifilter_name_arch(query);
+        if (relevant_installed_na.empty()) {
+            rpm::SolvQuery relevant_installed_n(all_installed);
+            relevant_installed_n.ifilter_name_arch(query);
```

Comment:
What is the meaning of the code?
Isn't the contents of `relevant_installed_n` the same as `relevant_installed_na`? Same source and same filters applied.

Reply:
Good point, the code must use `relevant_installed_n.ifilter_name(query);`

Reply:
Also the if bellow was not correct. Thanks for notest.

Reply:
The code should decide whether package was not installed or not installed for particular architecture

Reply:
Try to downgrade `libdnf.i686`.